### PR TITLE
Regl-based scattergl and scatterpolargl traces

### PR DESCRIPTION
**Pull request by etpinard**
This PR is a big one :muscle: :rocket: :rotating_light: 

About 99.99% of it is the fruit of @dfcreative's 6-month worth of work. @dfcreative rewrote our `scattergl` trace type using `regl`. In brief, `scattergl` traces are now drawn in a single canvas -- no matter how many subplots there are on the graphs -- that sits above cartesian SVG axes. Reusing the SVG axes logic fixed about 10 bugs by itself. Plotting in a single canvas improves performance for multi-subplot graphs immensely as well as removing those annoying _too many active WebGL contexts_ warnings. For more info on our new regl-based `scattergl` trace type, see https://github.com/plotly/plotly.js/pull/1869. For the sake of brevity, all that work was squashed into a single commit b64928d

The remaining 0.01% of this PR adds a new trace type: `scatterpolargl` which, as the name suggests, is a scatter trace type that'a plotted on polar subplots using WebGL. This new trace type doesn't bring much new code into the library, it simply combines logic from `scattergl` and `scatterpolar` on top of polar (SVG) subplots from #2200.

### CODE REVIEW

**Comment by etpinard** [comment]
... I'm fighting a few intermittent test failures on CI. I'll try to fix them the best I can.

Meanwhile, @dfcreative there's a few things I think we should add before merging:

- [ ] we should update the `gl2d_marker_symbol` mock to show that `scattergl` symbols are now on-par with `scatter`'s [marker_symbols](https://rreusser.github.io/plotly-mock-viewer/#marker_symbols) mock.
- [ ] we add a mock for `fill: 'toself'` cc https://github.com/plotly/plotly.js/issues/1334#issuecomment-356075940
- [ ] add a few `scatterpolargl` tests (I'll take care of this one)

File: src/traces/scattergl/index.js [hunk 1]
```diff
@@ -561,7 +561,7 @@ function sceneUpdate(container, subplot) {
             if(scene.fill2d) scene.fill2d.update(opts);
             if(scene.scatter2d) scene.scatter2d.update(opts);
             if(scene.line2d) scene.line2d.update(opts);
-            if(scene.error2d) scene.error2d.update([].push.apply(opts, opts));
+            if(scene.error2d) scene.error2d.update(opts.concat(opts));
```
Comment (etpinard) [inline src/traces/scattergl/index.js]: what does this fix?

File: src/traces/scattergl/index.js [hunk 2]
```diff
@@ -997,16 +997,21 @@ function hoverPoints(pointData, xval, yval, hovermode) {
...
+                Math.min(xl, xr), Math.min(ya._rl[0], ya._rl[1]),
+                Math.max(xl, xr), Math.max(ya._rl[0], ya._rl[1])
             );
         }
         else {
             ids = stash.tree.range(
-                xa.p2c(xpx - MAXDIST), ya.p2c(ypx + MAXDIST),
-                xa.p2c(xpx + MAXDIST), ya.p2c(ypx - MAXDIST)
+                Math.min(xl, xr), Math.min(yl, yr),
+                Math.max(xl, xr), Math.max(yl, yr)
```
Comment (etpinard) [inline src/traces/scattergl/index.js]: what does this fix?

File: src/traces/scattergl/index.js (hunk 1 above)
Comment (etpinard) [reply src/traces/scattergl/index.js]: I see it fixes https://github.com/plotly/plotly.js/issues/2259#issuecomment-358620568

Thanks.

**Comment by alexcjohnson** [comment]
@dfcreative @etpinard you get a whole troupe for this one
💃  💃  💃  💃 
💃  💃  💃  💃 
💃  💃  💃  💃 
💃  💃  💃  💃

**Pull request merged**